Shared/Samples/Sketch on map/SketchOnMapView.swift

```diff
@@ -144,7 +144,7 @@ extension GeometryEditorMenu {
             } label: {
                 Label("Delete Selected Element", systemImage: "xmark.square.fill")
             }
-            .disabled(model.selection == nil)
+            .disabled(model.selection == nil || !(model.selection?.canBeDeleted ?? false))
```

Comment:
@rolson, this precondition isn't documented on `deleteSelectedElement()`, should that be added?

Reply:
FYI: Instead of (essentially) checking for `nil` twice, it would be better to move the logic into a computed property:

```swift
var deleteButtonDisabled: Bool {
    guard let selection = model.selection else { return true }
    return !selection.canBeDeleted
}
```

Reply:
> @rolson, this precondition isn't documented on deleteSelectedElement(), should that be added?

Sure

Reply:
> @rolson, this precondition isn't documented on deleteSelectedElement(), should that be added?

Actually does it fail silently or crash? If crash, then yes it needs to be documented in GeometryEditor.

Reply:
@rolson It was crashing:

![capture](https://user-images.githubusercontent.com/16397058/230181980-0bb9b100-2875-4402-a31b-68c6057b3495.gif)